lib/core/src/Cardano/Wallet.hs

```diff
@@ -468,6 +471,40 @@ readWallet ctx wid = db & \DBLayer{..} -> mapExceptT atomically $ do
   where
     db = ctx ^. dbLayer @s @k
 
+
+data ErrFetchRewards
+    = ErrFetchRewardsNetwork ErrNetworkUnavailable
+    | ErrFetchRewardsNoSuchWallet ErrNoSuchWallet
+
+
+-- Rather than force all callers of 'readWallet' to wait for fetching the
+-- account balance (via the 'NetworkLayer'), we expose this function for it.
+fetchWalletRewardBalance
+    :: forall ctx s k t.
+        ( HasDBLayer s k ctx
+        , HasNetworkLayer t ctx
+        , HasRewardAccount s
+        )
+    => ctx
+    -> WalletId
+    -> ExceptT ErrFetchRewards IO (Quantity "lovelace" Word64) -- FIXME: Proper Error
+fetchWalletRewardBalance ctx wid = db & \DBLayer{..} -> do
+    let pk = PrimaryKey wid
+    cp <- liftE1 $ mapExceptT atomically $ withNoSuchWallet wid $ readCheckpoint pk
+    case rewardAccount $ getState cp of
+        Just acc -> liftE2 $ getAccountBalance nw acc
+        Nothing -> return $ Quantity 0
+  where
+    db = ctx ^. dbLayer @s @k
+    nw = ctx ^. networkLayer @t
+
+    liftE1 = withExceptT $ error "TODO: fetchReward liftE1"
+    liftE2 x = do
+        r <- liftIO $ runExceptT x
+        case r of
+            Right v -> return v
+            Left _ -> return $ Quantity 0
```

Comment:
**[change required]**

Do not silently swallow an error. If one error deserves to be ignore, use a clear pattern match to show it but avoid holes. 

**Why**

There are two reasons, the first one is:

- https://github.com/input-output-hk/cardano-wallet/wiki/Coding-Standards#prefer-pattern-matching-to-equality-testing-on-sum-types

And the second is that, if the account doesn't exist on the blockchain, we may want to default to a value of `0`. However, in case we get any sort of error, we should either fail, or, at the very least, log an error.
Note also that this is also an abuse of the `liftE2` function IMO which does much more than just lifting the error. I'd suggest moving the default handling logic to another function and to compose.